vector/src/vector.rs

```diff
@@ -0,0 +1,877 @@
+                if idx < data.len() {
+                    data.set(idx, elt);
+                } else {
+                    data.push_back(elt);
+                }
+            }
+            Node::Interior { children } => {
+                let bucket_idx = extract_index::<N>(idx, height);
+                assert!(height >= 1);
+                assert!(bucket_idx <= children.len());
+                if bucket_idx < children.len() {
+                    Rc::make_mut(&mut children[bucket_idx]).set(height - 1, idx, elt);
+                } else {
+                    let mut leaf = Chunk::new();
+                    leaf.push_back(elt);
+
+                    let mut child = Node::Leaf { data: leaf };
+                    for _ in 1..height {
+                        let mut children = Chunk::new();
+                        children.push_back(Rc::new(child));
+                        child = Node::Interior { children };
+                    }
+                    children.push_back(Rc::new(child));
+                }
+            }
+        }
+    }
+
+    /// Deletes and returns the last element of this subtree (which is assumed to be non-empty).
+    ///
+    /// Returns true if popping made this subtree empty.
+    fn pop(&mut self) -> (T, bool) {
+        match self {
+            Node::Leaf { data } => {
+                debug_assert!(!data.is_empty());
+                let ret = data.pop_back();
+                (ret, data.is_empty())
+            }
+            Node::Interior { children } => {
+                let (ret, child_empty) =
+                    Rc::make_mut(children.last_mut().expect("empty interior node")).pop();
+                if child_empty {
+                    children.pop_back();
+                }
+                (ret, children.is_empty())
+            }
+        }
+    }
+
+    /// Shrinks the length of this subtree to `len`.
+    ///
+    /// Assumes that the length is less than this node's current length.
+    fn truncate(&mut self, height: u8, len: usize) {
+        match self {
+            Node::Leaf { data } => {
+                debug_assert!(height == 0);
+                data.drop_right(len);
+            }
+            Node::Interior { children } => {
+                // If `len` is small enough, we may just want to drop some children
+                // and their entire subtrees.
+                let max_child_len = N.pow(u32::from(height));
+                let num_full_children = len / max_child_len;
+                let extra = len % max_child_len;
+                if extra > 0 {
+                    children.drop_right(num_full_children + 1);
+                    Rc::make_mut(&mut children[num_full_children]).truncate(height - 1, extra);
+                } else {
+                    children.drop_right(num_full_children);
+                }
+            }
+        }
+    }
+}
+
+/// [`Vector`] is a persistent vector (also known as a "bitmapped vector trie").
```

Comment:
Nitpick: I feel like this should be the module's documentation, and not `Vector`.